processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -376,32 +377,56 @@ private Assignment resolveViaMethod(Type sourceType, Type targetType, boolean co
             return null;
         }
 
+        /**
+         * Applies matching to given method only
+         *
+         * @param sourceType the source type
+         * @param targetType the target type
+         * @param method     the method to match
+         * @return an assignment if a match, given the criteria could be found. When the method is a
+         * buildIn method, all the bookkeeping is applied.
+         */
+        private Assignment applyMatching(Type sourceType, Type targetType, Method method) {
```

Comment:
Perhaps this should be named `resolveViaGivenMethod`? Or something in that direction. `applyMatching` differs from the other methods in my opinion

Reply:
not entirely.. We still have to apply all criteria to really figure out if it matches (qualifiers, xml-stuff, resultTypes, etc)

Reply:
OK